_episodes/08-reproducibility.md

```diff
@@ -0,0 +1,144 @@
+
+~~~
+docker save image/name:tag -o image-archive.tar
+~~~
+{: .bash}
+
+These tar images can become quite large and Zenodo supports uploads up to 50GB so you may need to compress your archive to make it fit on Zenodo using a tool such as gzip (or zip):
+
+~~~
+gzip image-archive.tar
+~~~
+{: .bash}
+
+Once you have your archive, you can [deposit it on Zenodo](https://zenodo.org/deposit/) and this will:
+
+   - Create a long-term archive snapshot of your Docker image which people (including your future self) can download and reuse or reproduce your work.
+   - Create a persistent DOI (*Digital Object Identifier*) that you can cite in any publications or outputs to enable reproducibility and recognition of your work.
+
+In addition to the archive file itself, the deposit process will ask you to provide some basic metadata to classify the image and the associated work.
+
+Note that Zenodo is not the only option for archiving and generating persistent DOIs for images. There are other services out there - for example, some organizations may provide their own, equivalent, service.
+
+## Reproducibility good practice
+
+   - Make use of images to capture the computational environment required for your work.
+   - Decide on the appropriate granularity for the images you will use for your computational work - this will be different for each project/area. Take note of accepted practice from contemporary work in the same area. What are the right building blocks for individual images in your work?
+   - Document what you have done and why - this can be put in comments in the Dockerfile and the use of the image desccribed in associated documentation and/or publications. Make sure that references are made in both directions so that the image and the documentation are appropriately linked.
+   - When you publish work (in whatever way) use an archiving and DOI service such as Zenodo to make sure your image is captured as it was used for the work and that is obtains a persistent DOI to allow it to be cited and referenced properly.
+
+## Container Granularity
+
+As mentioned above, one of the decisions you may need to make when containerising your research workflows
+is what level of *granularity* you wish to employ. The two extremes of this decision could be characterised
+as:
+
+  - Create a single container image with all the tools you require for your research or analysis workflow
+  - Create many container images each running a single command (or step) of the workflow and use them in
+    sequence
+
+Of course, many real applications will sit somewhere between these two extremes.
+
+> ## Positives and negatives
+> What are the advantages and disadvantages of the two approaches to container granularity for research
+> workflows described above? Think about this
+> and write a few bullet points for advantages and disadvantages for each approach in the course Etherpad.
+{: .challenge}
```

Comment:
Would be good to have an answer for this challenge, even if it is exhaustive of all the possible answers learners might come up with.